frontstage/views/account/account_transfer_survey.py

```diff
@@ -28,303 +26,143 @@
+    if request.method == "POST":
+        if not form.validate():
+            return render_template(
+                "surveys/surveys-transfer/recipient-email-address.html", form=form, errors=form.errors
+            )
+        party_id = session.get_party_id()
+        respondent_details = party_controller.get_respondent_party_by_id(party_id)
 
-@account_bp.route("/transfer-surveys/recipient-email-address", methods=["POST"])
-@jwt_authorization(request)
-def transfer_survey_post_email_entry(session):
-    form = AccountSurveyShareRecipientEmailForm(request.values)
-    party_id = session.get_party_id()
-    respondent_details = party_controller.get_respondent_party_by_id(party_id)
-    if not form.validate():
-        errors = form.errors
-        return render_template(
-            "surveys/surveys-transfer/recipient-email-address.html", session=session, form=form, errors=errors
-        )
-
-    if "emailAddress" in respondent_details:
         if respondent_details["emailAddress"].lower() == form.data["email_address"].lower():
             errors = {"email_address": ["You can not transfer surveys to yourself."]}
-            return render_template(
-                "surveys/surveys-transfer/recipient-email-address.html", session=session, form=form, errors=errors
-            )
-    flask_session["transfer_survey_recipient_email_address"] = form.data["email_address"]
-    return redirect(url_for("account_bp.send_transfer_instruction_get"))
+
+            return render_template("surveys/surveys-transfer/recipient-email-address.html", form=form, errors=errors)
+
+        flask_session["transfer_survey_recipient_email_address"] = form.data["email_address"]
+        return redirect(url_for("account_bp.send_transfer_instruction_get"))
+
+    return render_template("surveys/surveys-transfer/recipient-email-address.html", form=form)
 
 
 @account_bp.route("/transfer-surveys/send-instruction", methods=["GET"])
 @jwt_authorization(request)
-def send_transfer_instruction_get(session):
+def send_transfer_instruction_get(_):
     email = flask_session["transfer_survey_recipient_email_address"]
-    share_dict = {}
-    for business_id in flask_session["transfer_survey_data"]:
+    surveys_to_be_transferred = {}
+    surveys_to_transfer_map = flask_session["surveys_to_transfer_map"]
+
+    for business_id, survey_ids in surveys_to_transfer_map.items():
         selected_business = get_business_by_id(business_id)
         surveys = []
-        for survey_id in flask_session["transfer_survey_data"][business_id]:
+        for survey_id in survey_ids:
             surveys.append(survey_controller.get_survey(app.config["SURVEY_URL"], app.config["BASIC_AUTH"], survey_id))
-        share_dict[selected_business[0]["id"]] = {
+
+        surveys_to_be_transferred[selected_business[0]["id"]] = {
             "name": selected_business[0]["name"],
             "surveys": surveys,
         }
+        flask_session["transferred_surveys"] = surveys_to_be_transferred
```

Comment:
This isn't right, and means if a user breaks out a journey , the 2 todo page would show it without completing the transfer. This has been left in due to time pressure and will need to be fixed

Reply:
Agreed. Although it feels like the message on the todo page should be a permanent message, not a flash message from this session data, based on the presence of the "pending_surveys" for that "shared by" user.